### PR TITLE
Make :distribution:docker:check Gradle configuration cache compatible

### PR DESCRIPTION
**Pull request by breskeby**
fix cache compatibility issue in TestFixturesPlugin and support yamlRestTest in `:distribution:docker:yamlRestTest`

### CODE REVIEW

**Review by mark-vieira — APPROVED** [review]

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerUtils.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.docker;
+
+import org.elasticsearch.gradle.Architecture;
+
+public class DockerUtils {
+
+    public static boolean isArchitectureSupported(Architecture architecture, DockerSupportService service) {
```
Comment (mark-vieira) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerUtils.java:15]: Why don't we just add this as a method to `DockerSupportService` itself?

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/testfixtures/TestFixturesPlugin.java [hunk 2]
```diff
@@ -106,12 +105,13 @@ public void apply(Project project) {
             .register("postProcessFixture", TestFixtureTask.class, task -> {
                 task.getFixturesDir().set(testFixturesDir);
                 task.dependsOn(buildFixture);
-                configureServiceInfoForTask(
-                    task,
-                    project,
-                    false,
-                    (name, port) -> task.getExtensions().getByType(ExtraPropertiesExtension.class).set(name, port)
-                );
+                // configureServiceInfoForTask(
```
Comment (mark-vieira) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/testfixtures/TestFixturesPlugin.java:108]: Can this be removed?

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/testfixtures/TestFixturesPlugin.java [hunk 3]
```diff
@@ -131,12 +131,20 @@ public void apply(Project project) {
             return composePath != null ? composePath : "/usr/bin/docker-compose";
         }));
 
-        tasks.named("composeUp").configure(t -> {
+        tasks.withType(ComposeUp.class).named("composeUp").configure(t -> {
             // Avoid running docker-compose tasks in parallel in CI due to some issues on certain Linux distributions
             if (BuildParams.isCi()) {
                 t.usesService(dockerComposeThrottle);
```
Comment (mark-vieira) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/testfixtures/TestFixturesPlugin.java:137]: Can we remove this throttle now that we are effectively throttling these things on the docker support service?

File: distribution/docker/build.gradle [hunk 4]
```diff
@@ -188,26 +190,29 @@ ext.dockerBuildContext = { Architecture architecture, DockerBase base ->
...
 }
-
-def createAndSetWritable(Object... locations) {
-  locations.each { location ->
-    File file = file(location)
-    file.mkdirs()
-    file.setWritable(true, false)
-  }
-}
+//
```
Comment (mark-vieira) [inline distribution/docker/build.gradle:193]: Can this be removed?

File: distribution/docker/build.gradle [hunk 5]
```diff
@@ -488,11 +505,11 @@ for (final Architecture architecture : Architecture.values()) {
 
   addBuildEssDockerImageTask(architecture)
 }
-
-boolean isArchitectureSupported(Architecture architecture) {
-  Provider<DockerSupportService> serviceProvider = GradleUtils.getBuildService(project.gradle.sharedServices, DockerSupportPlugin.DOCKER_SUPPORT_SERVICE_NAME)
-  return serviceProvider.get().dockerAvailability.supportedArchitectures().contains(architecture)
-}
+//
```
Comment (mark-vieira) [inline distribution/docker/build.gradle:508]: Can we remove?

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/testfixtures/TestFixturesPlugin.java (hunk 3 above)
Comment (breskeby) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/testfixtures/TestFixturesPlugin.java:137]: I guess we can. will do

File: distribution/docker/build.gradle (hunk 5 above)
Comment (breskeby) [reply distribution/docker/build.gradle]: done

File: distribution/docker/build.gradle (hunk 4 above)
Comment (breskeby) [reply distribution/docker/build.gradle:193]: done

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/testfixtures/TestFixturesPlugin.java (hunk 2 above)
Comment (breskeby) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/testfixtures/TestFixturesPlugin.java]: done

**Pull request merged**